src/ripple/app/paths/impl/XRPEndpointStep.cpp

```diff
@@ -273,7 +283,20 @@ make_XRPEndpointStep (
     StrandContext const& ctx,
     AccountID const& acc)
 {
-    auto r = std::make_unique<XRPEndpointStep> (acc, ctx.isLast, ctx.j);
+    // For historical reasons, offer crossing is allowed to dig further
+    // into the XRP reserve than an ordinary payment.  (I believe it's
+    // because the trustline was created after the XRP was removed.)  Pass
+    // a flag that allows us to simulate the historical behavior.
+    //
+    // Note that reduced reserve only happens if the trustline does not
+    // currently exist.
+    bool reduceReserve = ctx.offerCrossing && ctx.isFirst;
+    if (reduceReserve)
+        reduceReserve = !ctx.view.read (keylet::line (acc, ctx.strandDeliver));
```

Comment:
This needs to go into the doc that lists the old offer crossing behaviors we are preserving, and then David will weigh in on all of them at once.